### ISSUE TITLE
FO4Edit "Fatal: Error loading plugin list: <EAccessViolation . . ." after update to Fallout 4 v1.7.90

### ISSUE THREAD

**Issue by archtrtd1** [TanninOne/modorganizer#917]
I decided to let Steam update my game from 1.6.9 to 1.7.9 yesterday.  Unfortunately this update has caused FO4Edit to throw an error/not launch while loading the plugins from my heavily modded profile.  I keep getting this message:  "Fatal: Error loading plugin list: EAccessViolation: Access violation at address 6946F20C in module 'MSVCR120.dll'. Read of address 01306000".

Operating system:  Windows 7 64 bit (fully up to date)
Mod Organizer version 2.0.7  (installed to C:\Users\me\AppData\Local\Fallout4)
FO4Edit version 3.1.3  (installed to C:\Games\Steam\Steamapps\common\Fallout 4)
Fallout 4 version 1.7.9.0 (installed to C:\Games\Steam\Steamapps\common\Fallout 4)
I have all of my executables (with the exception of Steam) set to "Run as Administrator".
I have added all of the executables to the "Excluded Processes" list in Microsoft Security Essentials.
I have no trouble actually launching/playing the game.

Things I have noticed during troubleshooting:
1. The addresses change on every new attempt to launch FO4Edit using my current playthrough's profile. 
2. I can launch FO4Edit directly from my Fallout 4 folder without errors.
3. I can launch FO4Edit using my "default" profile consisting of only the FO4 plugins and the Unofficial Patch without errors.
4. FO4Edit has never closed correctly for me.  It always crashes when I exit (although it still functions correctly when making edits/patches).

Here's what I have tried so far:
1. Uninstalling and Reinstalling VCRedistributable 2012 and 2013 (in the correct order of 32bit 2012, 64bit 2013 32bit 2013, 64bit 2013).
2. Uninstalling FO4Edit and reinstalling current version.
3. Uninstalling FO4Edit and reinstalling older versions.
4. Uninstalling and reinstalling Mod Organizer (backing up my profiles/mods).  This includes trying both install options of Mod Organizer (I can't remember the names right now), and to different locations on my C drive.
5. I have rebuilt my modded profile from scratch, manually, thinking that the .txt files were corrupted.
6. I have tried launching FO4Edit with F4SE disabled.
7. I have tried launching FO4Edit with enb disabled.
8. I have verified the game cache with Steam 3 times (nothing needed updating).
9. I have checked the Mod Organizer and Fallout 4 folders to confirm that they allow write access.
10. I have run sfc/scannow.

**Comment by Gernash** [comment]
Does this help?
http://wiki.step-project.com/User:Gernash/MODWiki#Setting_up_your_modding_environment

**Comment by archtrtd1** [comment]
I will try reinstalling Mod Organizer/FO4Edit to the locations suggested on the MODWiki tomorrow.

Perhaps someone can explain why  I would suddenly have to change the locations of these programs.  
I am a little frustrated to have to completely reinstall Mod Organizer/FO4Edit, again.  My game (with 140+ mods) runs well.  FO4Edit was running just fine until the 1.7.9 Fallout 4 update.

**Comment by Gernash** [comment]
You do not have to reinstall, if you are not using profiles (portable mode) then just profiles follow the directions then copy the folders that are in the MO2 folder into the profile folder QED

**Comment by Gernash** [comment]
I beat mine with a stick all the time its fine. only time I crash is if i change the contents of a mod folder and do a refresh, if i try and close MO2 it crashes with option to send debug report. I'd say a goo 80%+ of the issues people have here  is the Antivirus exclusions and the data being kept in the user\local data\modorganizer it just does not work and relocating it does not work you need the Symbolic link and leave the MO2 setting alone or nothing much works reliably.

**Comment by Gernash** [comment]
http://forum.step-project.com/topic/11094-wip-fallout-4-using-modorganizer-2/#entry176885

I wrote a Howto convert from portable to profile mode and retain all you stuff (if you need it)

**Comment by archtrtd1** [comment]
First of all, thank you Gernash for answering my post and sharing your wiki instructions with me.  Great work putting together all of that information in an easily accessible format.  I apologize if I seemed a little short in my previous post.  I've spent about 8 hours trying to fix this problem.  FO4Edit is mandatory for the way I mod.  I have some mods which require conflict resolution when they are updated.   I also have a few mods that I have manually merged their compatibility patches into (have to redo them every time those mods update).  So this means that I cannot play my current saves until this issue is resolved.  I have used MO for about 4 years (Oblivion, Fallout 3, Fallout NV, and Skyrim which all very heavily modded) and have NEVER encountered a problem this frustrating.

Well, I went ahead and reinstalled Mod Organizer per your instructions on the wiki step page.  The game launches correctly with all of my mods - so I know that I installed MO and my modded FO4 folders correctly (with the symbolic link established).  I then deleted all of the FO4Edit files from my Fallout 4 folder and reinstalled them to ModOrganizer2Data/Tools/FO4Edit.  I tried launching FO4Edit and am still getting the error:  "Fatal: Error loading plugin list: EAccessViolation: Access violation at address 6DE4F20C in module 'MSVCR120.dll'. Read of address 0103B000".  

I have no idea how to proceed from here.  There are very few posts regarding this issue and none of them explain how to fix this.  The post I found on the FO4Edit forum basically says "We do not support Mod Organizer users".

I do feel better not having my Mod Organizer files located in the users appdata directory.

**Comment by archtrtd1** [comment]
I think the problem may be the Plugins.txt.  
Here's the full errror:
Loading active plugin list: C:\Users\me\AppData\Local\Fallout4\Plugins.txt
Fatal: Error loading plugin list: EAccessViolation: Access violation at address 6DE4F20C in module 'MSVCR120.dll'. Read of address 015A1000

**Comment by archtrtd1** [comment]
I ran the vanilla launcher to see if it would regenerate the plugins.txt and now I cannot launch Mod Organizer.  I get the error "Failed to create 'Default'".

**Comment by Gernash** [comment]
Tes5edit not FO4Edit

**Comment by Gernash** [comment]
check if there is a folder called c:\program with nothing in it if so delete it

**Comment by Gernash** [comment]
inside the forums of the wiki there is somebody else I'm helping with a conversion all the info relates to you http://forum.step-project.com/topic/11094-wip-fallout-4-using-modorganizer-2/page-2#entry176977

**Comment by archtrtd1** [comment]
What do you mean "Tes5edit not FO4Edit"?  Actually isn't xEdit?  There is no c:\program folder.

**Comment by Gernash** [comment]
I was reading another forum and answered it here sorry for the confusion

**Comment by Gernash** [comment]
Here's the full errror:
Loading active plugin list: C:\Users\me\AppData\Local\Fallout4\Plugins.txt
??????
your errot should read
Loading active plugin list:  [whatever drive]C:\Users\me\AppData\Local\Fallout4\profiles\Default
is that a typo?

**Comment by archtrtd1** [comment]
No. I copied it directly from the error screen.  I had to delete the  <> symbols because this forum kept deleting the stuff contained within them.

**Comment by Gernash** [comment]
sooo I'd do regedit and search for AppData\Local\Fallout4\Plugins.txt and see what is doing it

Don't you love standalone install :)

**Comment by archtrtd1** [comment]
Ok. I have to reinstall Mod Organizer real quick, then I will search regedit.

**Comment by Gernash** [comment]
Because that looks to me to be a NMM registery issue linking to FO4Edit.

**Comment by archtrtd1** [comment]
Nope. I have never used Nexus Mod Manager.

**Comment by archtrtd1** [comment]
What am I supposed to be looking for in regedit?  There is no AppData subdirectory anywhere in the registry.  I opened all of the HKEY directories.

**Comment by Gernash** [comment]
okok
close MO2
go to C:\Users\me\AppData\Local\Fallout4
make a backup and delete the contents
start game through steam when you get to main menu exit
open MO2 go to download tab
find a download you already have installed run it and choose replace.
tru FO4edit

**Comment by archtrtd1** [comment]
Ok. Backed up/deleted the Fallout4 folder.  Ran Fallout through Steam.  The folder was recreated  (it contains DLCList.txt and Plugins.fo4viewsettings) .  I reinstalled The Beantown Interiors Project v8.3.  Still getting the FO4Edit crash.

**Comment by Gernash** [comment]
did you do it in the order listed

**Comment by archtrtd1** [comment]
Yes.

**Comment by Gernash** [comment]
Do you have any 3rd pary dll's in the Fallout 4 folder like F4SE? if so is there a F4SE folder with DLL's init?

**Comment by archtrtd1** [comment]
No.  My F4SE mods are installed/run through Mod Organizer.

**Comment by Gernash** [comment]
Also try logging out of the PC and back in as you many have a crashed app in the background

**Comment by Gernash** [comment]
the error points to something is read only, try right clicking the MO2 folder and MO2data folder and going to properties and unchecking the read only box and apply to all

**Comment by archtrtd1** [comment]
Restarted my computer. Still getting the error.

I tried this workaround:  https://github.com/TanninOne/modorganizer/issues/610
Manually adding in all of the plugins from my load order did not work either.  I did manually create a plugins.txt since it is missing.

I had a plugins.txt file yesterday, but deleted all of the files in the "C:\Users\me\AppData\Local\Fallout4" to try to refresh the files.  For the record, FO4Edit did not work for me when I did have the plugins.txt file, either.  It only contained the Fallout4.esm, DLCRobot.esm, DLCworkshop01.esm, DLCCoast.esm, DLCworkshop02.esm, and DLCworkshop03.esm names.

**Comment by Gernash** [comment]
Yea, it's pointing to the wrong place...
try ccleaner https://www.piriform.com/ccleaner/download

**Comment by Gernash** [comment]
did you reser the read only permissions on the 2 folders and apply to all?

**Comment by Gernash** [comment]
in the fo4edit log in the fo4edit folder does it look like this?
FO4Edit 3.1.3 EXPERIMENTAL starting session 2016-08-07 23:26:29
Using Fallout4 Data Path: C:\SteamLibrary\steamapps\common\Fallout 4\Data\
Using ini: j:\Users\me\Documents\My Games\Fallout4\Fallout4.ini
Using save path: j:\Users\me\Documents\My Games\Fallout4\Saves\
Using plugin list: C:\Users\me\AppData\Local\Fallout4\Plugins.txt
Using settings file: C:\Users\me\AppData\Local\Fallout4\Plugins.fo4viewsettings
Loading active plugin list: C:\Users\me\AppData\Local\Fallout4\Plugins.txt
[Fallout4.esm] Loading file

**Comment by archtrtd1** [comment]
What am I supposed to be cleaning with ccleaner?  I don't want to mess anything else up.

What two folders?

No.  Here is what it looks like:
FO4Edit 3.1.3 EXPERIMENTAL starting session 2016-08-09 01:10:27
Using Fallout4 Data Path: C:\Games\Steam\steamapps\common\Fallout 4\Data\
Using ini: C:\Users\me\Documents\My Games\Fallout4\Fallout4.ini
Using save path: C:\Users\me\Documents\My Games\Fallout4\Saves\
Using plugin list: C:\Users\me\AppData\Local\Fallout4\Plugins.txt
Using settings file: C:\Users\me\AppData\Local\Fallout4\Plugins.fo4viewsettings
Loading active plugin list: C:\Users\me\AppData\Local\Fallout4\Plugins.txt
Fatal: Error loading plugin list: <EAccessViolation: Access violation at address 6F39F20C in module 'MSVCR120.dll'. Read of address 01250000>

**Comment by Gernash** [comment]
can you actually go to C:\Users\me\AppData\Local\ModOrganizer\Fallout 4\profiles i.e. browse to it in explorer

**Comment by archtrtd1** [comment]
Yes.

**Comment by archtrtd1** [comment]
I don't understand why I can open FO4Edit with just the DLCs and the Unofficial Patch, but not my heavily modded profile.

**Comment by Gernash** [comment]
MO2 is sending the wrong virtual folder information via usvfs would be the technal answer I assume

**Comment by Gernash** [comment]
My next step would be to run the uninstaller from the MO2 Folder then run ccleaner then reboot then reinstall MO2

**Comment by Gernash** [comment]
As I can see no linking in the registry for MO2 and the Application they use make shure you delete all temp files and folders with ccleaner incase there is a randumb temp folder it uses and when you reinstall it goes back to it....

**Comment by Gernash** [comment]
Make sure you backup the Mo2data folder

**Comment by Gernash** [comment]
Another rudumb thaucht is you go into the spanner thingy in MO2 -->Pathc tab if you click the .... does it go to the correct folder?
base dir being C:/Users/me/AppData/Local/ModOrganizer/Fallout 4

**Comment by Gernash** [comment]
@TanninOne 
Where are the paths for the virtual links stored when you run a 3rd party app from MO2 this fault seems to be that it's pointing to the wrong spot.

**Comment by archtrtd1** [comment]
I thought my base directory was supposed to be:  C:/ModOrganizer2Data/Fallout 4

I do have a directory here:  "C:/Users/me/AppData/Local/ModOrganizer/Fallout 4"  Should I change the base directory to this one?

**Comment by Gernash** [comment]
yes

**Comment by Gernash** [comment]
base directory does not work hence the whole workaround with linked folders

**Comment by Gernash** [comment]
which is the point of the whole wiki....

**Comment by archtrtd1** [comment]
I followed the wiki and had Mod Organizer working (per my long post above).  Then I had this happen: "I ran the vanilla launcher to see if it would regenerate the plugins.txt and now I cannot launch Mod Organizer. I get the error "Failed to create 'Default'"."  
So I reinstalled Mod Organizer per the wiki page - but when I tried to re-establish the link, I received the notification that it was already there.

**Comment by archtrtd1** [comment]
I had Mod Organizer (installed to the default location) running just fine until the update.  Before I updated Saturday night, I was able to run FO4Edit with my modded profile.  I believe the update caused this problem but I cannot explain why or how it is affecting the MO/FO4Edit/modded profile relationship.  Like I said, I can launch a basic profile from Mod Organizer with FO4Edit containing only the base FO4 files and the unofficial patch.

**Comment by Gernash** [comment]
move the fo4 to somewhere else 
Start Mo2 
choose fallout 4
close fo4
copy the contents of your backed up fallout 4 dir into the newly created one
start mo2

**Comment by Gernash** [comment]
you are on the right track a few more clicks

**Comment by Gernash** [comment]
once this is done Mo2 will work as intended

**Comment by archtrtd1** [comment]
I moved my Fallout 4 folder (containing all of my profles, etc.)  Now I receive the error "failed to create Default".

**Comment by Gernash** [comment]
when you move the fallout 4 directory NOT rename it ovet to root folder or something then start MO2 as an admin you should get a splash screen click new fallout 4 etc

**Comment by archtrtd1** [comment]
I completely reinstalled Mod Organizer again.  I have been cleaning out my temp files from Windows, etc to make sure there isn't anything left over.

No matter what I do, it seems that I have to have a Fallout 4 folder located at C:/Users/me/AppData/Local/ModOrganizer/ Fallout4.  If there is nothing in the folder, I get prompted to select the type of install for Mod Organizer . . .

**Comment by Gernash** [comment]
that is fine it's pointing to teh MO2data folder
as intended

**Comment by Gernash** [comment]
it should have teh permissions 
system
your user
Administrators
all should be full control

**Comment by archtrtd1** [comment]
That is how I have everything set (I've had to do this for all of my modded games back to Oblivion).  I am the administrator and everything is set to run as administrator.  Everything has full write permissions.

**Comment by Gernash** [comment]
MKLINK cannot use to delete symbolic link. To remove a symbolic link, simply delete them as if you’re removing a normal file. Just make sure you don’t delete the original file.

**Comment by Gernash** [comment]
and you keep getting "failed to create Default".?

**Comment by archtrtd1** [comment]
Sorry.  I posted 6 posts above that "I completely reinstalled Mod Organizer again." Now Mod Organizer is running but I have to have this intact:  C:/Users/me/AppData/Local/ModOrganizer/ Fallout4.  I changed the paths of the base directory to ModOrganizer2Data and everything works correctly (except FO4Edit).

**Comment by archtrtd1** [comment]
I ran the uninstaller from the Mod Organizer folder before I reinstalled, of course.  Then deleted all of the residual folders. Then I remade everything again.

**Comment by Gernash** [comment]
symbolic link is a magic invisable link to another folder the C:/Users/me/AppData/Local/ModOrganizer should have a pic of a folder with an arrow the arrow symbolizes the link to tou mo2 folder the data in kept there not your app data folder
Hence the reason for the symbolic link
You can put data anywhere

**Comment by archtrtd1** [comment]
It does not have that symbol.  I get "cannot create a file when that file already exists" if I try to recreate the symbolic link.

**Comment by archtrtd1** [comment]
Are you running Windows 7?

**Comment by Gernash** [comment]
Win 10 64x Pro
C:/Users/me/AppData/Local/
ren ModOrganizer to ModOrganizer.old
Open an administrative command prompt, then create a symbolic link with the following command "mklink /J C:\Users\{User}\AppData\Local\ModOrganizer C:\ModOrganizer2Data" 
now you should see the ModOrganizer with the shorcut arrow
copy the contents of ModOrganizer.old into ModOrganizer

**Comment by archtrtd1** [comment]
That is exactly what I have done (three times). I followed your wiki page.   I get "cannot create a file when that file already exists" if I try to recreate the symbolic link in the cmd prompt (run as admin).

**Comment by Gernash** [comment]
windows key+ x
command prompt (admin)
delete C:\Users\me\AppData\Local\ModOrganize 
mklink /J C:\Users\me\AppData\Local\ModOrganizer C:\ModOrganizer2Data

**Comment by archtrtd1** [comment]
I ran the command prompt and received this message: 'delete' is not recognized as an internal or external command, operable program or batch file.

Should I use rmdir instead of delete?

**Comment by Gernash** [comment]
yea just remove del change the name what ever takes you fancy if all your files are in the back it up

**Comment by Gernash** [comment]
yes everyone else reading this I'm that patient

**Comment by Gernash** [comment]
you have like 30min left then I'm off to the movies.

**Comment by archtrtd1** [comment]
Sorry.  It took a while to copy my FO4 folder again.  I run an SSD and don't like a lot of clutter taking up space.

**Comment by archtrtd1** [comment]
Invalid switch - "J".

**Comment by Gernash** [comment]
not wanting to complicate the info but C:\ModOrganizer2Data can be anywhere you like
i.e. z:\the moon t:\harry the cat m:\Im going to the movies as long as the folder exists

mklink /J C:\Users\me\AppData\Local\ModOrganizer C:\ModOrganizer2Data

**Comment by archtrtd1** [comment]
I understand.  I am just following your instructions exactly to avoid confusion.  I used to have Mod Organizer installed to my C:\Games directory.

**Comment by Gernash** [comment]
2 folders
c:\Modorganizer2          The program lives here
c:\Modorganicer2data      The mods live here

**Comment by Gernash** [comment]
I dont have access to a win 7 just it's the same command
http://www.sevenforums.com/tutorials/278262-mklink-create-use-links-windows.html

**Comment by Gernash** [comment]
and you can copy then right click paste into the command window

**Comment by archtrtd1** [comment]
Thanks for your help, but this is not really getting anywhere.  I have followed your instructions to a 'T'.  Mod Organizer (along with the c:\Modorganizer2 and c:\Modorganizer2data folders) and the symbolic link were installed correctly.  MOD ORGANIZER BROKE when I ran Fallout 4 from the original Fallout4.exe.  I then had to reinstall everything to get it working again.  I have explained all of this in my previous posts.

**Comment by Gernash** [comment]
NP Hope you find a resolution Have a good night

**Comment by archtrtd1** [comment]
I keep getting "Invalid switch - "J" whenever I try to remove the link.

**Comment by archtrtd1** [comment]
For anyone still interested in this thread, I have figured out that my FO4Edit problems are not caused by where my Mod Organizer directory is located.  

For some reason, after updating to Fallout 4 v1.7.9, FO4Edit was crashing while loading all 136 of my plugins at one time.  I noticed that I could launch my default profile (consisting of the FO4 files plus the Unofficial Patch) with FO4Edit with no issues.  So I decided to experiment and see if there was something wrong with one of the mod plugins.

I started with all of my mod added plugins unchecked.  Then I would add in (check mark) 4 to 7 plugins and launch FO4Edit, let it run through its usual start up phase, then exit.  I was able to add all of my plugins back into FO4Edit this way.  FO4Edit still crashes at exit, but at least I can fix/patch my mods again.

I have no explanation as to why FO4Edit was exhibiting this behavior. It was extremely frustrating to find that the errors/bug report being generated by FO4Edit, literally sent Gernash and I down a road to nowhere trying to fix the problem.

I can't thank Gernash enough for taking the time to try and help me with this mysterious issue.  Gernash, you are a truly kind and patient person.  You are doing wonderful work with your wiki-step page.  I wish you the best in life!

**Comment by Gernash** [comment]
The issue is 
usvfs sees the directory path you have in Base Directory
Mod organizer sees C:/Users/{your username}/AppData/Local/ModOrganizer/Fallout 4 irrespective of what is in the base path

If you leave the base dir in  C:/Users/{your username}/AppData/Local/ModOrganizer/Fallout 4 your overwrite folder will not work causing fo4edit to crash.

The solution if you search through all the posts the people are/were using is to change the base dir, copy the contents of C:/Users/{your username}/AppData/Local/ModOrganizer/ then delete C:/Users/{your username}/AppData/Local/ModOrganizer.

But this still causes the loot and FO4Edit crash issue on close also reports of Fallout4.exe using unusual amounts of ram and when you exit the game not closing and staying in active memory.

The guide I posted if a proof of concept to demonstrate how to get around the issue. 
It 100% works.

But if it's to hard to follow then please stick to NMM.

**Comment by Scott-Liberation** [comment]
So after toying with MO2 in complete ignorance, ran into the same problem as first post in this thread, and discovered the excellent wiki by @Gernash   

Everything was going swimmingly until, while working on my own merge patch, I found a handful of entries in FunctionalDisplays-AID-VIS.esp that didn't incorporate unofficial patch changes.  So I changed them.  Then went back to having the same old problem.

And FINALLY, I found the cause (or at least, a cause):  if MO2 puts any FO4Edit backups in my Data tab, like for example in my Functional Display mod, having a file "FO4Edit Backups\FunctionalDisplays-AID-VIS.esp.backup.2016_10_11_18_34_26" then I get the FO4Edit error:

Fatal: Error loading plugin list: <EAccessViolation: Access violation at address 71C4F608 in module 'MSVCR120.dll'. Read of address 01203000>

and again LOOT crashes on startup.  But if I move all my mods' "FO4Edit Backups" out of my mods and say, onto my desktop, then everything goes back to peachy-keen.

I realize this issue is closed, but I didn't see any other issues on this specific topic.  Wasn't sure whether it would be appropriate to generate a new issue or if someone would prefer to re-open this one, assuming this cause can be reproduced.  I'm new to this GitHub business so not sure what the etiquette  is for such things. =)

**Comment by Gernash** [comment]
I'll post the info in the Wiki, Thanks!!!

I've disabled the Fo4edit backup.

MO2 truly hates any file changes with FO4edit and even if you make edits and then FO4Edit moves a file into the overwrite MO2 crashes.... Have to look for a setting to just overwite existing file......

**Comment by TDarkShadow** [comment]
Another interesting fact I noticed.
I 'optimised' the DLC ESM's with FO4Edit by removing 'Identical To Master' records and 'undelete and disable references',
I wanted to keep the original ESMs in the vanilla Fallout 4 Data folder and the 'optimised' ones in MO2.
However, the FO4Edit stops working if the DLC masters gets overriden during runtime.
To fix this, simply uncheck the optimised esms in MO2 or replace Fallout 4 default masters by your optimised ones.
